src/main/java/org/jitsi/impl/protocol/xmpp/colibri/ColibriConferenceImpl.java

```diff
@@ -943,4 +952,46 @@ public void release()
             }
         }
     }
+
+    public static class Stats {
+        private final Clock clock;
+
+        public Stats(Clock clock)
+        {
+            this.clock = clock;
+        }
+
+        public Stats()
+        {
+            this(Clock.systemUTC());
+        }
+
+        /**
+         * An average of the time it takes to make allocate channel requests
+         * to JVB.
+         */
+        private final RateStatistics allocateChannelsReqTimes =
+            new RateStatistics((int)Duration.ofMinutes(1).toMillis());
+
+        private final RateStatistics healthCheckReqTimes =
+            new RateStatistics((int)Duration.ofMinutes(1).toMillis());
+
+        /**
+         * Notify the stats object how long an allocate channels request took
+         * to execute
+         * @param nanos the time, in nanoseconds
+         */
+        void allocateChannelsRequestTook(long nanos) {
+            allocateChannelsReqTimes.update((int)nanos, clock.millis());
+        }
+
+        @SuppressWarnings("unchecked")
+        public JSONObject toJson()
+        {
+            JSONObject json = new JSONObject();
+            json.put("avg_allocate_channels_req_time_nanos", allocateChannelsReqTimes.getRate());
```

Comment:
Hm, actually, `RateStatistics` doesn't do what we need here. This just returns the accumulated values (multiplied with a constant scale). We want the accumulated values divided by number of events.

Reply:
🤦 Yeah, this will need some more thought on what we want to use.